### ISSUE TITLE
SPDX 2→3 conversion: (old) Directional Relationships

### ISSUE THREAD

**Issue by puerco** [spdx/spdx-3-model#356]
### TL:DR

SPDX 3 should keep `containedBy`, `dependencyOf`, `describedBy`, `generatedFrom`, `prerequisiteFor` as valid, non-recommended relationship types for compatibility.

### Problem:

In the new model, relationships have clear `to:` and `from:` attributes, meaning the relationship vocabulary is now smaller. as flipped relationships are not needed anymore.

For example, in SPDX 2.3 we had `CONTAINS` and `CONTAINED_BY`, both are now expected to be replaced by a single `contains`. 

This sounds great on paper until you start converting older data and processes to SPDX3. While flipping `CONTAINED_BY` sounds easy enough, it fundamentally alters the SBOM graph when converting from 2 to 3. Tools written to understand the relationship interactions between elements need to get new logic to handle the new graph and conversion between 2.3 and 3 will be inevitably lossy.

### Proposal:

I think we should keep the older "flipped" relationship types as a compatibility crutch between SPDX 2.3 and 3. Their use can be discouraged in the documentation and annotated as planned to be deprecated in future versions but it solves a structural problem when converting between 2.3 and 3.

In order to have a lossless conversion of relationship data, we would need to keep only five (5) relationship types for compatibility:

| SPDX 2 | SPDX 3 Equivalent | Status |
| --- | --- | --- |
| CONTAINS | contains | :heavy_check_mark: in spec |
| CONTAINED_BY | N/A | Propose `containedBy` for compatibility | 
| DEPENDS_ON | dependsOn | :heavy_check_mark: in spec |
| DEPENDENCY_OF | N/A | Propose `dependencyOf` for compatibility |
| DESCRIBES	| describes | :heavy_check_mark: in spec |
| DESCRIBED_BY | N/A | Propose `describedBy` for compatibility |
| GENERATES | generates | :heavy_check_mark: in spec |
| GENERATED_FROM | N/A | Propose `generatedFrom` for compatibility |
| HAS_PREREQUISITE	 | prerequisite | :heavy_check_mark: in spec |
| PREREQUISITE_FOR | N/A | Propose `prerequisiteFor` for compatibility |

I've put together a [spreadsheet mapping the relationship types between SPDX 2.3 and 3](https://docs.google.com/spreadsheets/d/1y3cpmaB7H4RsLnyjGiGDRife7V1v0oJasgnsh3M3gNk/edit#gid=0) in case anybody finds it useful.

**Comment by goneall** [comment]
Talked with Kate and she will take this on as part of the relationship cleanup work.

**Comment by goneall** [comment]
@kestewart - checking if this is now resolved

**Comment by kestewart** [comment]
This has been resolved with the work that @JPEWdev and I have done.    Discussed this with @goneall and he is not aware of anything key outstanding at this point.    

If there are things overlooked,  please open more specific issue.

**Comment by goneall** [comment]
@puerco - here's the spreadsheet Kate was using for the analysis: https://docs.google.com/spreadsheets/d/1yb49TkcR_dVEH_k_iOcBjNZbYDjkmd0HVweIuDpS9eg/edit
